server/api/src/rsDataObjTrim.cpp

```diff
@@ -54,42 +56,76 @@ namespace
         return info;
     }
 
-    int get_time_of_expiration(const char* age_kw) {
+    auto get_time_of_expiration(const char* age_kw) -> std::uint64_t
+    {
         if (!age_kw) {
             return 0;
         }
+
+        const auto now = std::time(nullptr);
+        if (-1 == now) {
+            THROW(SYS_INTERNAL_ERR, fmt::format("{}: Error occurred getting current time.", __func__));
+        }
+
         try {
-            const auto age_in_minutes = std::atoi(age_kw);
-            if ( age_in_minutes > 0 ) {
-                const auto now = time(0);
-                const auto age_in_seconds = age_in_minutes * 60;
-                return now - age_in_seconds;
+            const auto age_in_minutes = std::stol(age_kw);
+            if (age_in_minutes < 0) {
+                THROW(SYS_INVALID_INPUT_PARAM,
+                      fmt::format("{}: {} value [{}] must be a non-negative value.", __func__, AGE_KW, age_kw));
             }
-            return 0;
+
+            constexpr decltype(age_in_minutes) seconds_in_one_minute = 60;
+            const auto age_in_seconds = age_in_minutes * seconds_in_one_minute;
+            if (0 != age_in_minutes && age_in_seconds / age_in_minutes != seconds_in_one_minute) {
+                // If the provided time was so large that representing it in seconds caused the value to wrap, just
+                // return 0 because the replicas would have to be older than epoch time 0 in order to be old enough to
+                // be trimmed. iRODS itself has not existed that long, so none of the replicas are old enough to be
+                // trimmed based on the input.
+                return 0;
+            }
```

Comment:
Is the comment still true if the replicas were registered and existed before iRODS?

Reply:
Can Linux systems handle times from before epoch time 0? If they can, and the iRODS catalog can, then we need to handle this.

If that is the case, we need to decide if such handling is worth the effort given that this feature is deprecated as of this PR.

Reply:
Some chaos here...
    https://stackoverflow.com/a/31709796

> For a signed 32-bit integer, the values range from 1901-12-13 to 2038-01-19 03:14:07 UTC

and then also...

> Add to this lack of specificity the fact that [a couple dozen other epochs](https://en.wikipedia.org/wiki/Epoch_(computing)#Notable_epoch_dates_in_computing) have been used by various software systems, some extremely popular and common. Examples include January 1, 1601 for NTFS file system & COBOL, January 1, 1980 for various FAT file systems, January 1, 2001 for Apple Cocoa, and January 0, 1900 for Excel & Lotus 1-2-3 spreadsheets.

And this crazy take...

> I recommend against tracking date-time as a count-from-epoch. Instead use specific data types where available in your programming language or database.

Reply:
I forgot the age option was being deprecated. It's fine as is.